src/lib/oper_data.c

```diff
@@ -596,24 +630,61 @@ void process_node(const struct lysc_node *s_node, json_object *json_obj, uint16_
 }
 
 /**
- * Converts JSON array objects to a YANG data tree based on the provided schema node.
- * This function is a wrapper around `process_node` for handling JSON arrays.
- * @param [in] json_array_obj: JSON array object to convert.
- * @param [in] top_node: Top-level schema node for mapping.
- * @param [in] lys_flags: Flag value used to filter out schema nodes containers and leafs
+ * Starts the porcessing of module schema, it processes every node in the schema to lyd_node if the node name
+ * is found in the input json_obj.
+ * First it looks for show commands to apply as it iterates through the schema node, then it stores the outputs
+ * into json_obj, and starts processing the schema nodes based on the content of that json_obj.
+ * @param [in] s_node: Schema node to be processed.
+ * @param [in] json_obj: JSON object containing the data to process.
+ * @param [in] lys_flags: flag value used to filter out schema nodes containers and leafs.
  *                        Use LYS_CONFIG_R to allow parsing read-only containers and leafs (for loading operational data).
  *                        Use LYS_CONFIG_W to allow parsing write containers and leafs (for loading configuration data).
- * @param [out] data_tree: Pointer to store the resulting YANG data tree.
+ * @param [in, out] parent_data_node: Pointer to the parent data node in the YANG data tree (lyd_node).
  */
-void jarray_to_lyd(json_object *json_array_obj, const struct lysc_node *top_node,
-                   uint16_t lys_flags, struct lyd_node **data_tree)
+void process_schema(const struct lysc_node *s_node, uint16_t lys_flags,
+                    struct lyd_node **parent_data_node)
 {
-    const struct lysc_node *node;
-    LY_LIST_FOR(top_node, node)
-    {
-        // Start with level 0 for top-level nodes, data_tree = NULL
-        process_node(node, json_array_obj, lys_flags, data_tree);
+    char *show_cmd = NULL;
+    const struct lysc_node *new_cmd_child;
+    struct json_object *cmd_output = NULL;
+    /* Create top-level lyd_node */
+    if (*parent_data_node == NULL) { // Top-level node
+        lyd_new_inner(NULL, s_node->module, s_node->name, 0, parent_data_node);
+    }
+
+    if (get_lys_extension(OPER_CMD_EXT, s_node, &show_cmd) == EXIT_SUCCESS) {
+        if (show_cmd == NULL) {
+            fprintf(stderr,
+                    "%s: ipr2cgen:oper-cmd extension found but failed to "
+                    "get the command value for node \"%s\"\n",
+                    __func__, s_node->name);
+            return; //review this needs to exit the whole thing not just single node processing
+        }
+        if (apply_ipr2_cmd(show_cmd)) {
```

Comment:
this if statement is confusing, please do if `(apply_ipr2_cmd(show_cmd) != EXIST_SUCCESS)`,
`apply_ipr2_cmd(show_cmd)` should return `EXIST_SUCCESS` or `EXIT_FAILURE` instead of sysrepo return code.